linux/usage/systemd.md

```diff
@@ -1,57 +1,103 @@
 
-On your Pi, create a .service file for your service, for example:
+## Service definitions
+`Systemd` defines each service in a separate file. Under `systemd` services are a type of 'unit': units are system resources under the control of `systemd`. The basic format of a service definition is as follows:
 
-myscript.service
+```
+[Unit]
+Description=...
+...
+
+[Service]
+...
+
+[Install]
+...
+```
+
+User services should be placed in `/etc/systemd/user`: the filename dictates the service name. For example, a file named `monitoring.service` would define the service named `monitoring`.
+
+### Example - one-time service
+One-time services are used when you wish to run a program which does some work, then exits: this is in contrast to a service which stays running all the time. Consider the following service definition:
+
+```
+[Unit]
+Description=Run my program once at boot time
+Wants=network-online.target
+After=network-online.target
+
+[Service]
+Type=oneshot
+ExecStart=/home/pi/myprogram.sh
+
+[Install]
+WantedBy=multi-user.target graphical.target
+```
+
+Let's say the above file is `/etc/systemd/user/myprogram.service`: this defines a new one-shot service named `myprogram`. The `myprogram` service will wait until `local-fs.target` and `network-online.target` have been reached, then run `/home/pi/myprogram.sh`.
+
+### Example - simple service
+A simple service runs continuously for as long as its dependencies are met. We define `/etc/systemd/user/netmusicplayer.service` as follows:
 
 ```
 [Unit]
-Description=My service
-After=network.target
+Description=Network music player
+Wants=network-online.target
+After=network-online.target
+Requires=mpd.target
 
 [Service]
-ExecStart=/usr/bin/python3 -u main.py
-WorkingDirectory=/home/pi/myscript
-StandardOutput=inherit
-StandardError=inherit
-Restart=always
-User=pi
+Type=simple
+ExecStart=/home/pi/netmusicplayer.sh
+ExecStop=/home/pi/netmusicplayer.sh stop
 
 [Install]
-WantedBy=multi-user.target
+WantedBy=multi-user.target graphical.target
 ```
-So in this instance, the service would run Python 3 from our working directory `/home/pi/myscript` which contains our python program to run `main.py`. But you are not limited to Python programs: simply change the ExecStart line to be the command to start any program/script that you want running from booting.
 
-Copy this file into `/etc/systemd/system` as root, for example:
+Our `netmusicplayer` service is started up using the script `/home/pi/netmusicplayer.sh`. We can also pass the `stop` parameter to that same script to stop `netmusicplayer`. Because the network music player uses [`mpd`](https://www.musicpd.org/), we use the `Requires=` directive to specify that `mpd` must be running before `netmusicplayer` starts up.
```

Comment:
"We can also pass the `stop` parameter to that same script to stop `netmusicplayer`." implies that the user themselves would pass the `stop` parameter to the shellscript, where I guess the intention is that the user would use `sudo systemctl stop netmusicplayer` as explained later? Perhaps the sentence just needs to be reworded to not use "We" ?

Reply:
Fixed